scripts/create_local_env_for_cil_development.sh

```diff
@@ -1,47 +1,94 @@
-#  -*- coding: utf-8 -*-
+#!/usr/bin/env bash
+
 # Copyright 2020 United Kingdom Research and Innovation
 # Copyright 2020 The University of Manchester
-    
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
-    
 #     http://www.apache.org/licenses/LICENSE-2.0
-    
 # Unless required by applicable law or agreed to in writing, software
 # distributed under the License is distributed on an "AS IS" BASIS,
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-    
 # Authors:
 # CIL Developers, listed at: https://github.com/TomographicImaging/CIL/blob/master/NOTICE.txt
-#! /bin/bash
 
-while getopts hn:p:e: option
- do
- case "${option}"
- in
-  n) numpy=${OPTARG};;
-  p) python=${OPTARG};;
-  e) name=${OPTARG};;
+set -euxo pipefail
+numpy='1.24'
+python='3.10'
+name=cil
+test_deps=0
+cil_ver=''
+while getopts hn:p:e:tv: option ; do
+  case "${option}" in
+  n) numpy="${OPTARG}" ;;
+  p) python="${OPTARG}" ;;
+  e) name="${OPTARG}" ;;
+  t) test_deps=1 ;;
+  v) test_deps=1 ; cil_ver="${OPTARG}" ;;
   h)
-   echo "Usage: $0 [-n numpy_version] [-p python version] [-e environment name]"
-   exit 
-   ;;
+    echo "Usage: $0 [-n numpy_version] [-p python_version] [-e environment_name] [-t] [-v cil_version]"
+    echo 'Where `-t` installs test dependencies, and `-v cil_version` implies `-t`'
+    exit
+    ;;
   *)
-   echo "Wrong option passed. Use the -h option to get some help." >&2
-   exit 1
-  ;;
- esac
+    echo "Wrong option passed. Use the -h option to get some help." >&2
+    exit 1
+    ;;
+  esac
 done
 
-echo Numpy $numpy
-echo Python $python
-echo Environment name $name
+echo "Numpy $numpy"
+echo "Python $python"
+echo "Environment name $name"
+
+conda_args=(create --name="$name"
+  python="$python"
```

Comment:
Is there anyway for this to automatically stay up to date with https://github.com/TomographicImaging/CIL/blob/ff4741092b7f10bde5e0572c0ac103288a163ffc/recipe/meta.yaml#L54-L70

Currently we don't have the ipp restrictions here.

Reply:
there's no easy way to keep a bash script in sync.
Maybe we should just use an environment.yaml file and ask people to substritute/`sed` numpy & python versions if they want.

Reply:
a yaml file we'd have to remember to update too I suppose? I think it's worth having one as well/instead however.

Reply:
we already have a `scripts/requirements-test.yml` file for this purpose. It just doesn't have:

- pinned/overrideable `numpy` & `python` versions (not solvable, just needs documenting IMO)
- distinction between build & runtime deps (could split into 2 yaml files?)
- sync with `recipe/meta.yaml` (any ideas?)
- sync with `docs/docs_environment.yml`
- I haz no idea wat `Wrappers/Python/environment.yml` is